src/Angular-CSharp.csproj.in

```diff
@@ -46,14 +48,14 @@
     </Exec>
     <Error Condition="'$(ErrorCode)' != '0'" Text="Node.js is required to build and run this project. To continue, please install Node.js from https://nodejs.org/, and then restart your command prompt or IDE." />
     <Message Importance="high" Text="Restoring dependencies using 'npm'. This may take several minutes..." />
-    <Exec WorkingDirectory="$(SpaRoot)" Command="npm install" />
+    <Exec WorkingDirectory="$(SpaRoot)" Command="$(SpaInstallCommand)" />
   </Target>
   <!--/+:cnd:noEmit -->
 
   <Target Name="PublishRunWebpack" AfterTargets="ComputeFilesToPublish">
     <!-- As part of publishing, ensure the JS resources are freshly built in production mode -->
-    <Exec WorkingDirectory="$(SpaRoot)" Command="npm install" />
-    <Exec WorkingDirectory="$(SpaRoot)" Command="npm run build -- --configuration production" />
+    <Exec WorkingDirectory="$(SpaRoot)" Command="$(SpaInstallCommand)" Condition="'$(SkipSpaBuild)' != 'True'" />
+    <Exec WorkingDirectory="$(SpaRoot)" Command="$(SpaBuildCommand)" Condition="'$(SkipSpaBuild)' != 'True'" />
```

Comment:
Can we use Condition="'$(DOTNET_RUNNING_IN_CONTAINER)' == 'True'" here instead? That will automatically do the right thing here without having to set anything additional.

Reply:
That will break users that have this setup as a single-stage build on an image with .NET and nodejs (https://github.com/dotnet/spa-templates/pull/58#issuecomment-1175746873).

I know some customers build .NET SPA apps this way.

If this should work automagically we could skip the SPA build when `DOTNET_RUNNING_IN_CONTAINER` **and** nodejs is not in the image.

Reply:
> and no nodejs.

Can you clarify what do you mean by that?

The appeal of `DOTNET_RUNNING_IN_CONTAINER` is that the container image already sets it, so the user doesn't have to take/make an additional step in their Dockerfile and is one less concept they need to understand.

Reply:
A user that wants to do a single-stage SPA build will have a .NET image (which should have `DOTNET_RUNNING_IN_CONTAINER`) that also includes `nodejs`.

If our logic is `SkipSpaBuild = DOTNET_RUNNING_IN_CONTAINER == "True"`, then the single-stage build will not work.

To continue supporting the single-stage SPA build, we need to extend the condition with: _and no nodejs_.

Reply:
> With what you suggest, after the change, users that want to do single-stage builds have to update their csproj.

They would have to update their docker image either way to pass an additional flag, wouldn't they? I am not against adding an extra flag, I just want to make sure we reason through the scenarios.

Not all the scenarios are weighted the same, and I lean way more towards making building against our images work without having to pass additional settings than making it easier for people building their own images, as I think the first scenario will be far more common.

Whenever we tell people to do this in their docker image, the additional step only involves adding an extra build step and copying the outputs to the final build, there is no need for them to go and add a flag to their dotnet publish step.